### PR TITLE
Feat: Enhance SSO Role Mapping with JMESPath

### PR DESCRIPTION
**Pull request by neelneelpurk**
## Description
This PR introduces robust role mapping capabilities using JMESPath expressions in the SSO configuration. It specifically addresses the need to map user roles based on complex conditions and standard OIDC claims.

### SSO & Role Mapping
- **Configured `OIDC_ROLE_ATTRIBUTE_PATH` or via `sso.oidc.roleAttributePath` in the `config.yaml`**:
  ```jmespath
  (contains(['...', ...], email) && 'admin' || "custom:team" == 'editor' && 'editor' || 'user'
  ```

### Testing
- **Added Unit Tests**:
  - `TestSSOService_HandleOIDCCallback_RoleMapping` in `sso_service_test.go` covering:
    - Simple and nested attribute mapping.
    - Custom attribute conditions (`custom:team`).
    - Handling of missing attributes.
  - Updated test setup to mock OIDC provider responses and prevent external network calls.

### Security
- **Dependency Updates**: Upgraded `golang.org/x/crypto` to address a known vulnerability.

### Limitations
- **Role Mapping**: Mapping roles based on group membership (e.g., `cognito:groups`) is currently **not implemented/supported** as this information is not returned in the standard OIDC `userinfo` response. Currently, role mapping relies on attributes available in the `userinfo` endpoint. Will add this feature via another PR.

## Verification
- **Automated Tests**:
  - Ran `go test -v ./services/...` - All tests passed, confirming the JMESPath logic handles edge cases correctly.

### CODE REVIEW

File: crossview-go-server/models/user.go [hunk 1]
```diff
@@ -2,9 +2,10 @@ package models
 
 import (
 	"crypto/rand"
+	"fmt"
 	"strings"
 	"time"
-	"fmt"
```
Comment (MoeidHeidari) [inline crossview-go-server/models/user.go]: as we can see "fmt" was already there. change is just removing and adding it again.
